lib/axiom/fs/stream/skeleton_file_system.js

```diff
@@ -0,0 +1,582 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AxiomError from 'axiom/core/error';
+import Ephemeral from 'axiom/core/ephemeral';
+import Path from 'axiom/fs/path';
+import SkeletonStdio from 'axiom/fs/stream/skeleton_stdio';
+import SkeletonReadableStream from 'axiom/fs/stream/skeleton_readable_stream';
+import SkeletonWritableStream from 'axiom/fs/stream/skeleton_writable_stream';
+
+/** @typedef Channel$$module$axiom$fs$stream$channel */
+var Channel;
+
+/** @typedef {ExecuteContext$$module$axiom$fs$base$execute_context} */
+var ExecuteContext;
+
+/** @typedef FileSystem$$module$axiom$fs$base$file_system */
+var FileSystem;
+
+/** @typedef {OpenContext$$module$axiom$fs$base$open_context} */
+var OpenContext;
+
+/** @typedef ReadableStream$$module$axiom$fs$stream$readable_stream */
+var ReadableStream;
+
+/** @typedef {StatResult$$module$axiom$fs$stat_result} */
+var StatResult;
+
+/** @typedef Stdio$$module$axiom$fs$stdio */
+var Stdio;
+
+/** @typedef WritableStream$$module$axiom$fs$stream$writable_stream */
+var WritableStream;
+
+/**
+ * Expose a file system implementation over a Channel.
+ *
+ * @constructor @extends {Ephemeral}
+ * @param {!string} description
+ * @param {!FileSystem} fileSystem
+ * @param {!Channel} channel
+ */
+export var SkeletonFileSystem = function(description, fileSystem, channel) {
```

Comment:
Just contemplating naming for this.  This class doesn't really qualify as _is-a_ `FileSystem`.  Maybe `FileSystemProxy`?  `FileSystemTransport`?  Maybe `SkeletonFileSystem` is fine.  We just seem to have a loose definition already (with `FileSystemManager` extends `FileSystem`, etc).